public/app/plugins/datasource/loki/components/LokiQueryField.tsx

```diff
@@ -5,14 +5,16 @@ import LokiLanguageProvider from '../language_provider';
 
 type LokiQueryFieldProps = Omit<
   LokiQueryFieldFormProps,
-  'syntax' | 'syntaxLoaded' | 'onLoadOptions' | 'onLabelsRefresh' | 'logLabelOptions'
+  'syntax' | 'syntaxLoaded' | 'onLoadOptions' | 'onLabelsRefresh' | 'logLabelOptions' | 'absoluteRange'
 >;
 
 export const LokiQueryField: FunctionComponent<LokiQueryFieldProps> = props => {
-  const { datasource, absoluteRange, ...otherProps } = props;
+  const { datasource, range, ...otherProps } = props;
+  const absoluteTimeRange = { from: range!.from!.valueOf(), to: range!.to!.valueOf() };
```

Comment:
`range!.from!` is this just a problem with the typings or is it really possible for it to be undefined?

Reply:
It is just a typing issue. We are passing range to `LokiQuertField` everywhere it is used. Wasn't sure if I should use `non-null operator`, or give "OR solution". But this seemed a better solution, as "OR solution" would never been used.

Reply:
Lets just add a comment here to mention that range.from is actually never optional.